### PR TITLE
fix infinite loop when generate unresolvable import

### PR DESCRIPTION
**Pull request by favadi**
when executing "generate" if an import package is unresolvable, gunk
creates infinite number of gunpkg-*.go files until terminated.
This happens because a recursive calls in Loader.Loop method that
occurs when check.Files returns an error.

Solution: terminate loader.Load if an error is reported.

### CODE REVIEW

File: testdata/scripts/loader_files.txt [hunk 1]
```diff
@@ -13,7 +13,7 @@ cmp p2/p2.gunk p2/p2.gunk.golden
 
 # loading files from different packages with types should error
 ! gunk generate p1/p1.gunk p2/p2.gunk
-stderr 'package p2; expected p1'
+stderr '-: gunk package name mismatch: "p1" "p2"'
```
Comment (mvdan) [inline testdata/scripts/loader_files.txt]: hmm, why is this changed?

File: loader/loader.go [hunk 2]
```diff
@@ -349,6 +349,10 @@ func (l *Loader) parseGunkPackage(pkg *GunkPackage) {
 		pkg.ProtoName = pkg.Name
 	}
 
+	if len(pkg.Errors) > 0 {
+		return
```
Comment (mvdan) [inline loader/loader.go]: shouldn't we report the errors? if they are caught elsewhere, just mention that in a comment here.

Comment (favadi) [reply loader/loader.go]: comment added.

File: testdata/scripts/loader_files.txt (hunk 1 above)
Comment (favadi) [reply testdata/scripts/loader_files.txt]: because with the change in loader.go, we now catch/return this error earlier.

File: testdata/scripts/generate_unexisting_import.txt [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# make sure go modules is turned off, and GOPATH is set to a nonexistent directory
```
Comment (mvdan) [inline testdata/scripts/generate_unexisting_import.txt]: Just noticed "unexisting" in this file name - the correct term would be "nonexistent"

Comment (mvdan) [reply testdata/scripts/generate_unexisting_import.txt]: Or even better, "missing" :)

Comment (favadi) [reply testdata/scripts/generate_unexisting_import.txt]: > Just noticed "unexisting" in this file name - the correct term would be "nonexistent"
 
changed all to nonexistent.
